tiingo/api.py

```diff
@@ -155,6 +166,67 @@ def get_ticker_price(self, ticker,
         else:
             return response.content.decode("utf-8")
 
+    def _build_url(self, stock, startDate, endDate, frequency):
+        url = "https://api.tiingo.com/tiingo/"
+        url += "daily/{}/prices?".format(stock)
+        if startDate is not None and endDate is None:
+            url += "&startDate={}".format(startDate)
+        if startDate is not None and endDate is not None:
+            url += "&startDate={}&endDate={}".format(startDate, endDate)
+        url += "&format=json&resampleFreq={}&token={}".format(frequency, self._api_key)
```

Comment:
Hey All,

Thanks for bringing this to my attention. Given the data in the API is read-only and static, I am not too concerned about leaking quite yet (SSL is also enforced server side). Once personalized data is opened via API (e.g. users being allowed to pull in their financial data from Tiingo), the authentication method is spec'd to be stricter (v2 of the API is being spec'd out to allow for reading of user data). 

Appreciate the work you two have done on this - I know from E-mails many people are finding this to be incredibly helpful

Reply:
Hey All

Thanks for bringing this to my attention. The server enforces HTTPS, meaning the API token in the URL is encrypted regardless. 

Therefore the security issue is if a user makes an error and accidentally exposes their token by publicly posting a URL. Users have done token resets because they've exposed their token in the Headers as well, so I'm not certain removing the token GET parameter would reduce human error. I think it's something all developers can relate to (accidentally committing a key to a repo). 

I am not too worried about this user error given the only data right now that would be exposed is market data, which presents no PII risk to the user. So if there is user error, it presents a business risk to Tiingo, not the user. I think we are OK assuming this risk for the convenience the <b>token</b> GET parameter gives people. 

You two are doing wonderful work - its greatly appreciated by myself and I know the community.